user_guide/annotations.rst

```diff
@@ -0,0 +1,118 @@
+Annotations
+===========
+
+Historically Behat used doc-block annotations instead of attributes to define steps, hooks and
+transformations in PHP contexts. These annotations are still available and you can use them instead
+of PHP attributes in your projects.
+
+Step Annotations
+----------------
+
+Here is an example of how you can define your steps using annotations:
+
+.. code-block:: php
+
+    // features/bootstrap/FeatureContext.php
+
+    use Behat\Behat\Context\Context;
+
+    class FeatureContext implements Context
+    {
+        /*
+         * @Given we have some context
+         */
+        public function prepareContext()
+        {
+            // do something
+        }
+
+        /*
+         * @When event occurs
+         */
+        public function doSomeAction()
+        {
+            // do something
+        }
+
+        /*
+         * @Then something should be done
+         */
+        public function checkOutcomes()
+        {
+            // do something
+        }
+    }
+
+The pattern that you would include as an argument to the step attribute should be listed here
+after the annotation, separated by a space
+
+Hook Annotations
+----------------
+
+Here is an example of how you can define your hooks using annotations:
+
+.. code-block:: php
+
+    // features/bootstrap/FeatureContext.php
+
+    use Behat\Behat\Context\Context;
+    use Behat\Testwork\Hook\Scope\BeforeSuiteScope;
+    use Behat\Behat\Hook\Scope\AfterScenarioScope;
+
+    class FeatureContext implements Context
+    {
+        /*
+         * @BeforeSuite
+         */
+        public static function prepare(BeforeSuiteScope $scope)
+        {
+        }
+
+        /*
+         * @AfterScenario @database
+         */
+        public function cleanDB(AfterScenarioScope $scope)
+        {
+        }
+    }
+
+Transformation Annotations
+--------------------------
+
+Here is an example of how you can define your transformations using annotations:
+
+.. code-block:: php
+
+    // features/bootstrap/FeatureContext.php
+
+    use Behat\Behat\Context\Context;
+
+    class FeatureContext implements Context
+    {
+        /*
+         * @Transform /^(\d+)$/
+         */
+        public function castStringToNumber($string)
+        {
+            return intval($string);
+        }
+    }
+
+Existing code
+-------------
+
+Even though annotations are still available, they will probably be deprecated and eventually removed in the future.
+In this way, for new projects we recommend that you don't use them. If your current project uses annotations, we
+recommend that you refactor your code to use PHP attributes instead. This can be done manually but you should be able
+to use the search and replace capabilities of your IDE to do this in a more automated way.
+
+Alternatively you may want to use a tool like `Rector`_ which can do automated refactoring of your code. Rector includes
```

Comment:
I think you could just say `Rector 2.0 includes a rule...` even if this is ready before the release is out. That's enough for people to work out whether to try it with an RC / wait for it IMO and saves waiting or coming back to this later.